Add warning about TRUNCATE transaction safety (#20983)
yugabyte/yugabyte-db: docs/content/preview/develop/best-practices-ysql.md

## TRUNCATE tables instead of DELETE

[TRUNCATE](../../api/ysql/the-sql-language/statements/ddl_truncate/) deletes the database files that store the table data and is much faster than [DELETE](../../api/ysql/the-sql-language/statements/dml_delete/) which inserts a _delete marker_ for each row in transactions that are later removed from storage during compaction runs.

{{<warning>}}

> The TRUNCATE command in the current implementation is not transactional. Also TRUNCATE is not MVCC-safe just as in Postgres. See [TRUNCATE](../../api/ysql/the-sql-language/statements/ddl_truncate#truncate) for more details.

Reviewer: ```suggestion
Currently, TRUNCATE command is not transactional. Also, TRUNCATE is not MVCC-safe similar to PostgreSQL. For more details, see [TRUNCATE](../../api/ysql/the-sql-language/statements/ddl_truncate#truncate). 
```
